p2p/src/authenticated/discovery/actors/listener.rs

```diff
@@ -103,16 +103,44 @@ impl<E: Spawner + Clock + Network + Rng + CryptoRng + Metrics, C: Signer> Actor<
         mut tracker: UnboundedMailbox<tracker::Message<C::PublicKey>>,
         mut supervisor: Mailbox<spawner::Message<SinkOf<E>, StreamOf<E>, C::PublicKey>>,
     ) {
+        // Track the rejection reason from the bouncer
+        let rejection_reason = std::sync::Arc::new(std::sync::Mutex::new(None));
+        let rejection_reason_clone = rejection_reason.clone();
+
         let (peer, send, recv) = match listen(
             context,
-            |peer| tracker.acceptable(peer),
+            |peer| {
+                let mut tracker = tracker.clone();
```

Comment:
this is jank but technically this the way you'd need to do this given the interface of `listen`